### PR TITLE
ci: add GitHub Actions CI

### LINKED ISSUE #18: Add linter check to CI

### LINKED ISSUE #53: ci: testing
We don't have CI for testing. We need to add it and fix the tests:

```
Ran 28 tests in 40.542 seconds, 26 succeeded, 2 failed

=========================================================
Failed tests:

exponential_backoff_test.test_default_wait_factor
exponential_backoff_test.test_success
```

See also:

https://github.com/tarantool/sharded-queue/pull/48

### PR DESCRIPTION
**Pull request by oleg-jukovec**
The patch adds CI for GitHub Actions and updates flaky tests.

Closes #53
Closes #18

### CODE REVIEW

**Review by GRISHNOV — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          tarantool-version: ${{ matrix.tarantool }}
+
+      - name: Clone the module
+        uses: actions/checkout@v3
+
+      - name: Install requirements
+        run: make deps
+
+      - run: make lint
```
Comment (better0fdead) [inline .github/workflows/tests.yml:41]: It would be nice to add names to steps.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      - run: make test
```
Comment (better0fdead) [inline .github/workflows/tests.yml:43]: It would be nice to add names to steps.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+      - run: make coverage
```
Comment (better0fdead) [inline .github/workflows/tests.yml:45]: It would be nice to add names to steps.

**Review by better0fdead — CHANGES_REQUESTED** [review]
It also would be nice to split commit into 2. 
Add CI and fix flaky tests.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (oleg-jukovec) [reply .github/workflows/tests.yml]: Added.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (oleg-jukovec) [reply .github/workflows/tests.yml]: Added.

**Review by better0fdead — APPROVED** [review]

**Pull request merged**